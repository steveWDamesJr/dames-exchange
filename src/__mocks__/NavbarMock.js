import React, { useState, useEffect } from 'react';

const Navbar = () => {
  const [setActiveMenu] = useState(true);
  const [screenSize, setScreenSize] = useState(undefined);

  useEffect(() => {
    const handleResize = () => setScreenSize(window.innerWidth);

    window.addEventListener('resize', handleResize);

    handleResize();

    return () => window.removeEventListener('resize', handleResize);
  }, []);

  useEffect(() => {
    if (screenSize <= 800) {
      setActiveMenu(false);
    } else {
      setActiveMenu(true);
    }
  }, [screenSize]);

  return (
    <div className="nav-container">
      <div className="logo-container">
        <div className="logo-dollar" size="large" />
        <div><a to="/">Welcome Cryptonians!</a></div>
        <button type="button">Click me</button>
      </div>
    </div>
  );
};

export default Navbar;
