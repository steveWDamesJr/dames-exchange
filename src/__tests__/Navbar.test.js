import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import NavbarMock from '../__mocks__/NavbarMock';

const Navbar = () => <>Home</>;

test('should display the nav text Home', () => {
  render(<Navbar />);
  screen.debug();
});

it('renders correctly', () => {
  const tree = renderer.create(
    <React.StrictMode>
      <Router>
        <Provider store={store}>
          <NavbarMock />
        </Provider>
      </Router>
    </React.StrictMode>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
