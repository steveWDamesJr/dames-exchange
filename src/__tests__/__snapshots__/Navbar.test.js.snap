// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="nav-container"
>
  <div
    className="logo-container"
  >
    <div
      className="logo-dollar"
      size="large"
    />
    <div>
      <a
        to="/"
      >
        Welcome Cryptonians!
      </a>
    </div>
    <button
      type="button"
    >
      Click me
    </button>
  </div>
</div>
`;
